lib/rules/require-default-prop.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @fileoverview Require default value for props
+ * @author Michał Sajnóg <msajnog93@gmail.com> (http://github.com/michalsnik)
+ */
+'use strict'
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Require default value for props',
+      category: 'Best Practices',
+      recommended: false
+    },
+    fixable: null,  // or "code" or "whitespace"
+    schema: []
+  },
+
+  create: function (context) {
+    // ----------------------------------------------------------------------
+    // Helpers
+    // ----------------------------------------------------------------------
+
+    /**
+     * Checks if the passed prop is required
+     * @param {Property} prop - Property AST node for a single prop
+     * @return {boolean}
+     */
+    function propIsRequired (prop) {
+      const propRequiredNode = prop.value.properties
+        .find(p =>
+          p.key.name === 'required' &&
+          p.value.type === 'Literal' &&
+          p.value.value === true
+        )
+
+      return Boolean(propRequiredNode)
+    }
+
+    /**
+     * Checks if the passed prop has a defualt value
+     * @param {Property} prop - Property AST node for a single prop
+     * @return {boolean}
+     */
+    function propHasDefault (prop) {
+      const propDefaultNode = prop.value.properties
+        .find(p => p.key.name === 'default')
+
+      return Boolean(propDefaultNode)
+    }
+
+    /**
+     * Finds all props that don't have a default value set
+     * @param {Property} propsNode - Vue component's "props" node
+     * @return {boolean}
+     */
+    function findPropsWithoutDefaultValue (propsNode) {
+      return propsNode.value.properties
+        .filter(prop => {
+          if (prop.value.type !== 'ObjectExpression') return true
```

Comment:
In case of spread operator `value` is going to be undefined:
`TypeError: Cannot read property 'type' of undefined`
```js
export default {
  props: {
    a: {
      ...test
    }
  }
}
```